TripCheckins/Authorization/FoursquareAuthorizer.swift

```diff
@@ -14,9 +14,26 @@ protocol FoursquareAuthorizerDelegate: class {
 
 final class FoursquareAuthorizer {
     
-    private let clientId = "SO0SBDKGOYSZCH4JMQOEHJHMNTETYOWCURGWAZ22BPHKQDWE"
-    private let clientSecret = "KBYTRDYUXSUULW4P4YDELCDQVF3FZ1ZKNHLJOF0MLZ2CGUMQ"
-    private let callbackURIString = "tripcheckins://foursquare"
+    private let clientId: String
+    private let clientSecret: String
+    private let callbackURIString: String
+    
+    convenience init?() {
+        guard let plistPath = Bundle.main.path(forResource: "foursquare_api_configuration", ofType: "plist"),
```

Comment:
I think it will be better to have a non optional init here and have an assert to make sure Foursquare is configured properly. 

Non-configured Foursquare auth flow should be solved by the developer during the development time and that way you won't need to handle the misconfigured service in the UI.

Reply:
I was thinking about that but my thinking was that it is not unrecoverable error. It means Foursquare can't be used for fetching checkins - so then I think it is up to the app coordinator to decide what to do. At the moment it doesn't make much sense to not have 4square connected (I can add the assert one level up?) but I kinda don't want to leave that decision to the authorizer. What do you think?

Reply:
Hmm... I think you need to make sure that the developer is aware that the FoursquareAuthorizer is not configured properly.

If it's a recoverable error, how about throwing? You can still catch it and do what needs to be done in the app coordinator but it will be clearer to the developer that it's due to a bad configuration.